File: 1829/solution.py
class Solution:
    def getMaximumXor(self, nums: list[int], k: int) -> list[int]:
        xor = 0
        for num in nums:
            xor ^= num
        res = []
        complement = pow(2, k) - 1
        for i in range(len(nums) - 1, -1, -1):
            res.append(xor ^ complement)
            xor ^= nums[i]
        return res

File: 1829/solution.java
class Solution {
    public int[] getMaximumXor(int[] nums, int k) {
        int xor = 0;
        for (int num : nums) {
            xor ^= num;
        }
        int biggestNum = (int) Math.pow(2, k) - 1;
        int[] res = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            res[i] = xor ^ biggestNum;
            xor ^= nums[nums.length - i - 1];
        }
        return res;
    }
}

// time O(n)
// space O(1)